moon/controller.py

```diff
@@ -211,10 +213,19 @@ def on_joint_position(self, data):
         pass
 
     def on_rot(self, data):
-        temp_yaw, self.pitch, self.roll = [normalizeAnglePIPI(math.radians(x/100)) for x in data]
```

Comment:
well "on_rot" is already deprecated and you should get "orientation" (quaternion), which you can parse into Euler angles if needed. Note, that median on angles is not good near the singularity (yes, it is definitely better then average), but reading 0, 1, 359, 358, 120 will return 120, which is wrong. On the other hand filtering on quaterion could work better (no, I do not know details at the moment).

Reply:
Good point and something certainly to pay attention to in some near future iteration (eg remove on_rot everywhere, replace by properly filtered quaternions) but probably good enough for now as is?